proc/proc.go

```diff
@@ -113,15 +120,29 @@ func resolveCriticalSignal() syscall.Signal {
 	return syscall.SIGUSR2
 }
 
-func sendSignal(sig syscall.Signal, ps []*os.Process) error {
-	merrs := &MultiErrors{}
-	for _, p := range ps {
-		if err := p.Signal(sig); err != nil {
-			merrs.es = append(merrs.es, err)
-		}
+type OsProcess struct {
+	process *os.Process
+}
+
+func NewOsProcess(p *os.Process) OsProcess {
+	return OsProcess{
+		process: p,
 	}
-	if len(merrs.es) == 0 {
-		return nil
+}
+
+func (p OsProcess) Pid() int {
+	return p.process.Pid
+}
+
+func (p OsProcess) Signal(s os.Signal) error {
+	return p.process.Signal(s)
+}
+
+func (p OsProcess) MemoryUsagePercent() (uint64, error) {
+	limit, usage, err := mem.LimitAndUsageForProc(p.process)
+	if err != nil {
+		return 0, err
 	}
-	return merrs
+
+	return (usage * 100) / limit, nil
```

Comment:
I think that this has been copied from the original implementation but, as we are already here, can we somehow return an error if `limit` is zero ? Just for sake of not dividing by zero.

Reply:
Fixed